src/utils.rs

```diff
@@ -1,8 +1,28 @@
-use ethers_core::types::{U256, U64};
+use ethers_core::types::{U256, U64, H160};
 use serde::{de, Deserialize};
 use serde_json::Value;
 use std::str::FromStr;
 
+pub fn deserialize_h160<'de, D>(deserializer: D) -> Result<H160, D::Error>
+where
+    D: de::Deserializer<'de>,
+{
+    Ok(match Value::deserialize(deserializer)? {
+        Value::String(s) => {
+            if s.as_str() == "0x" {
+                return Ok(H160::zero());
+            }
+
+            H160::from_str(s.as_str()).map_err(de::Error::custom)?
+        }
+        Value::Number(num) => H160::from_low_u64_ne(
+            num.as_u64()
+                .ok_or_else(|| de::Error::custom("Invalid number"))?,
+        ),
+        _ => return Err(de::Error::custom("wrong type")),
+    })
```

Comment:
As you pointed out, a number cannot be returned in place of an address; so we should remove this arm.